### PR TITLE
Convert Menu Header component from class based to function based component

### LINKED ISSUE #24763: Convert ./components/widgets/menu/menu_header.tsx from Class Component to Function Component
We want to move most of our codebase from Class Components to Function Components. This decision is based on [React recommendations](https://react.dev/reference/react/Component).

The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring.

Here are a few examples of the required changes:
https://github.com/mattermost/mattermost/pull/24187
https://github.com/mattermost/mattermost/pull/24210
https://github.com/mattermost/mattermost/pull/24183
https://github.com/mattermost/mattermost/pull/24181

If the tests do not work because they rely on the behavior of Class Components they also have to be migrated to functionaly test the same behavior. We prefer the use of `react-testing-library` for new tests, but it is not required.

When submitting the PR, please make sure to identify where and when the component shows, for QA to properly test the changes.

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA LINK: MM-54799

### PR DESCRIPTION
**Pull request by dipandhali2021**
#### Summary
This PR converts the MenuHeader component from class based component to functional based component

#### Ticket Link
Fixes #24763

#### Release Note
```
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/widgets/menu/menu_header.tsx [hunk 1]
```diff
@@ -14,17 +14,12 @@ type Props = {
 /**
  * @deprecated Use the "webapp/channels/src/components/menu" instead.
  */
-export default class MenuHeader extends React.PureComponent<Props> {
-    public render() {
-        const {children, onClick} = this.props;
+const MenuHeader: React.FC<Props> = ({ children, onClick }) => {
```
Comment (jespino) [inline webapp/channels/src/components/widgets/menu/menu_header.tsx:17]: nit: It is more consistent with the examples to use directly the Props type in the parameters directly, something like:
```suggestion
const MenuHeader = ({ children, onClick }: Props) => {
```

File: webapp/channels/src/components/widgets/menu/menu_header.tsx [hunk 2]
```diff
@@ -14,17 +14,12 @@ type Props = {
...
-            <li
-                className='MenuHeader'
-                onClick={onClick}
-            >
-                {children}
-            </li>
-        );
-    }
-}
+export default MenuHeader;
```
Comment (jespino) [inline webapp/channels/src/components/widgets/menu/menu_header.tsx:25]: It was a pure component so it is expected to have here a `React.memo` to make it memoized.
```suggestion
export default React.memo(MenuHeader);
```

**Review by jespino — CHANGES_REQUESTED** [review]
A couple of improvements needed, but in general looks good to me. Thanks @dipandhali2021! :sparkles:

**Comment by dipandhali2021** [comment]
Hey, @jespino .Thanks for reviewing the PR and suggestions. I have committed new changes and suggestion.

**Comment by larkox** [comment]
@dipandhali2021 Have you signed the CLA? You can see in this comment the details: https://github.com/mattermost/mattermost/pull/24799#issuecomment-1751642447

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
@dipandhali2021 You can see in the checks on the PR that there are a few test and linter errors. The linter errors can be almost sure auto fixed, so you just have to run `npm run fix` in the channels folder. Also, the tests seem to be just snapshot tests, so you should be able to fix those running `npm run test -- -u -- path/to/test/file.test.tsx`

**Comment by dipandhali2021** [comment]
Hey, @larkox . I am sorry to close this PR. Previously I didn't setup properly mattermost locally due to which `npm run fix` was not working. But now I have setup mattermost properly. I will submit the PR by today only. 
Sorry for the inconvenience provided.

**Comment by larkox** [comment]
No problem at all :smile: Just wanted to make sure if you dropped working on it so we could get the ticket back to up for grabs. But if you plan to keep working on it, all is fine :smile:

**Pull request closed without merging**